lib/tournament.ex

```diff
@@ -12,67 +29,75 @@ defmodule Tournament do
   """
   @spec tally(input :: list(String.t())) :: String.t()
   def tally(input) do
-    initial_output = "Team                           | MP |  W |  D |  L |  P"
-
     input
-    |> Enum.reduce(%{}, fn line, state ->
-      [team1, team2, outcome] = parse_line(line)
-      team1_tally = Map.get(state, team1, %{mp: 0, wins: 0, losses: 0, draws: 0, points: 0})
-      team2_tally = Map.get(state, team2, %{mp: 0, wins: 0, losses: 0, draws: 0, points: 0})
-
-      case outcome do
-        "win" ->
-          state
-          |> update_tally(team1, apply_team(team1_tally, "win"))
-          |> update_tally(team2, apply_team(team2_tally, "loss"))
-
-        "loss" ->
-          state
-          |> update_tally(team1, apply_team(team1_tally, "loss"))
-          |> update_tally(team2, apply_team(team2_tally, "win"))
-
-        "draw" ->
-          state
-          |> update_tally(team1, apply_team(team1_tally, "draw"))
-          |> update_tally(team2, apply_team(team2_tally, "draw"))
-      end
+    |> Stream.map(&parse_input_line_values/1)
+    # TODO: Convert to Stream.transform/3
+    |> Enum.reduce(%{}, fn [team1, team2, outcome], state ->
+      update_tallies(state,
+        get_team_tally(state, team1),
+        get_team_tally(state, team2),
+        outcome
+      )
     end)
-    |> Enum.to_list()
     |> Enum.sort_by(fn {_key, %{points: points}} -> points end, :desc)
-    |> Enum.reduce(initial_output, fn {team, tally}, acc ->
-      Enum.join([acc, "\n", team_line(team, tally)])
+    |> Enum.reduce(headline(), fn {team, tally}, acc ->
+      Enum.join([acc, "\n", team_stats_to_formatted_line(team, tally)])
     end)
   end
 
-  def parse_line(line) do
-    String.split(line, ";")
-  end
+  defp parse_input_line_values(input), do: String.split(input, @separator_input)
 
-  def team_line(team, %{mp: mp, wins: wins, draws: draws, losses: losses, points: points}) do
-    [
-      String.pad_trailing(team, 30),
-      String.pad_leading("#{mp}", 2),
-      String.pad_leading("#{wins}", 2),
-      String.pad_leading("#{draws}", 2),
-      String.pad_leading("#{losses}", 2),
-      String.pad_leading("#{points}", 2)
-    ]
-    |> Enum.join(" | ")
-  end
+  defp headline(), do:
```

Comment:
This replaces `initial_output =` followed by a manually typed out number of spaces in a static string. The idea is to build the head of the output the same way we generate each line, so we're sure the spacing is robust.